### ISSUE TITLE
July 2022: Meetup in Cologne

### ISSUE THREAD

**Issue by kawogi** [Rustaceans/rust-cologne#96]
# ToDo

- [X] Settle for a date and location: **Wednesday 2022-07-06, 19:15 CEST** at C4 (default: first Wednesday each month)
- [ ] Send "Save the Date" info (with link to this ticket)
- [X] Basic Agenda (default: Open Space)
- [x] Final invitation (deadline **2022-06-29**)
  - [x] [meetup](https://www.meetup.com/rustcologne/events/286890733/)
  - [x] [website](https://rust.cologne/2022/07/06/physical-reboot.html)
  - [x] [twitter](https://twitter.com/RustCologne/status/1542543911352242176)
  - [ ] [community calendar]

# Invitation draft

```markdown
Title: Physical Reboot

Dear Rustaceans,

our next Meetup will be on **Wednesday 2022-07-06 19:15 CEST**.

After a very long break we're happy to announce our next Meetup held in person at the C4!

Matthias will give a **talk about WASM**:
> WebAssembly is neither "web" nor "assembly" — but 100% amazing. It is our chance to build a fast, safe, and universal  compilation target for all languages and all platforms. Let me show you why this is cool and how to get started with WebAssembly in Rust; we will also briefly touch on WASI.

Kai will perform some **beginner level interactive coding** implementing a Digital Rain effect in the Terminal (as seen in the Movie _The Matrix_). We'll learn how idiomatic Rust looks like and how `clippy` can teach you to become a better rustacean over time.

Any remaining time will be used as **open space**: Imagine a bazaar, where you can suggest topics and spontaneously join any discussion you find interesting. It all depends on your interests. You can enjoy some drinks, meet nice people and discuss about Rust.

**[You can register here](...)**.

We are looking forward to seeing you. :-)

Yours,
Florian and Kai
―  
The meetup will likely be held in German, we will however reevaluate this at the beginning of the evening and may switch to English if needed.
―
Due to COVID-19 still being around, bringing a face mask will be necessary. For the same reason we need to know how many people will attend the meetup. Please make sure to register and, in case you are no longer able to attend, cancel your registration. Thank you.
```
# Rust News
- [Rust 1.62](https://github.com/rust-lang/rust/blob/stable/RELEASES.md#version-1620-2022-06-30)
  + `#[derive(Default)]` on enums
  + `cargo add`
- [RLS is being deprecated](https://blog.rust-lang.org/2022/07/01/RLS-deprecation.html)
- c2rust is back: https://immunant.com/blog/2022/06/back/
- Building a web application with Rust and WebAssembly https://kerkour.com/web-application-with-rust-and-webassembly

**Comment by kawogi** [comment]
Topics suggested in the recent (virtual) meetup:

- WASM (WebAssembly)
- Tokio
- Tauri (https://github.com/tauri-apps/tauri)
- Writing Terminal Applications
- SIMD
- Procedural Macros

For beginners:

- clippy
- writing an actix based webserver
- iterators
- unit tests

Maybe we could also fetch some open entry on rosetta code and do a live coding
https://rosettacode.org/wiki/Reports:Tasks_not_implemented_in_Rust

Corona measures (those need to be integrated in the invitation):
- open windows
- bring a face mask (we may ask you to wear it if we feel the room gets too crowded)

Other things that could be part of the meetup:

- present something random from This Week In Rust even if it doesn't seem important

I'll try to come up with a topic to be able to complete the invitation. Suggestions are still welcome.

**Comment by mre** [comment]
Are you planning the meetup for July or August? It says "2022-07-06" but the final invitation date is after that on "2022-07-29", so I wonder what date is correct now.
If you plan to do the meetup in July I could give a talk about WASM. I might not be here in August however.

**Comment by kawogi** [comment]
Thanks for noticing. I fixed the invitation date to match the meetup next week. For August I'll have to skip as well.

WASM talk would be great.
I plan to do a live coding for implementing the "Matrix Digital Rain" effect in the terminal. That should be beginner friendly, yet entertaining. Maybe we feel like over-engineering the result after that ;)
(https://rosettacode.org/wiki/Matrix_digital_rain https://en.wikipedia.org/wiki/Matrix_digital_rain)

**Comment by mre** [comment]
Okay, thanks for the clarification. I'd be delighted to do the WASM talk then. Here's an abstract if you need one for the announcement. 😊 

> WebAssembly is neither "web" nor "assembly" — but 100% amazing. It is our chance to build a fast, safe, and universal compilation target for all languages and all platforms. Let me show you why this is cool and how to get started with WebAssembly in Rust; we will also briefly touch on WASI.

**Comment by kawogi** [comment]
Thanks for noticing. I fixed the invitation date to match the meetup next week. For August I'll have to skip as well.

WASM talk would be great.
I plan to do a live coding for implementing the "Matrix Digital Rain" effect in the terminal. That should be beginner friendly, yet entertaining. Maybe we feel like over-engineering the result after that ;)
(https://rosettacode.org/wiki/Matrix_digital_rain https://en.wikipedia.org/wiki/Matrix_digital_rain)

**Comment by kawogi** [comment]
@Florob I updated the invitation draft. Can you have a look if this is ok to be released?

**Comment by Florob** [comment]
@kawogi Looked mostly fine to me. I made a few small changes, fixing grammar and (I hope) improving text flow. Also I think reminding people registration is mandatory this time around is about as important as reminding them to cancel.

**Comment by kawogi** [comment]
@Florob I created the Event on Meetup. Any last words before I announce that officially?

**Comment by kawogi** [comment]
I added a new post to the Website but it isn't showing after waiting 10 minutes. Anything I forgot? Am I too impatient?

On a side note: We might want to mention those topics in the "What's new" slot of our meetup (found in recent TWIR).

- c2rust is back: https://immunant.com/blog/2022/06/back/
- Building a web application with Rust and WebAssembly https://kerkour.com/web-application-with-rust-and-webassembly
- stabilized explicit_generic_args_with_impl_trait https://github.com/rust-lang/rust/issues/83701
- stabilized scoped threads https://rust-lang.github.io/rfcs/3151-scoped-threads.html

**Comment by Florob** [comment]
@kawogi: Meetup looks good! Feel free to announce it.

Website is indeed strange. github-pages deploy fails without error message (build succeeds). Github seems a bit wonky right now anyway though. I'll re-trigger it again later and hope it works then.

We should collect "What's new" items in the issue so we have one place to look at during meetup. Rust 1.62 release is also coming up today, that will give us some things to talk about.
I'm not sure to what extend we should talk about things that have "only" landed in nightly. E.g. I knew about scoped threads for a while, but decided it would be more apt to talk about them once released in Rust 1.63.

**Comment by appelgriebsch** [comment]
Link to Euro Rust conference: https://eurorust.eu/

**Comment by kawogi** [comment]
As promised I uploaded the code to [github](https://github.com/kawogi/matrix-digital-rain) and [rosetta code](https://rosettacode.org/wiki/Matrix_digital_rain#Rust). I added some comments to make it easier to follow.

The VSCode plugins I've been using were:

- rust-analyzer
- even better toml
- Cargo
- crates
- intent rainbow
- rainbow brackets

Thank you all for coming and your feedback. It's been a great revival! See you next time.
